### PR TITLE
Add Unit Tests for All Classes and Decorators

### PR DESCRIPTION
**Pull request by Zafron047**
### Dear Reviewer,
Please review this pull request and provide your feedback or approve to merge to incorporate the unit tests, which enhance the reliability and maintainability of the codebase.

## Description
This pull request addresses the project requirements by adding unit tests for all classes and decorators, as specified in the project's requirements. The following classes and decorators have been covered:

- Book
- Person
- Student
- Teacher
- Classroom
- Rental
- BaseDecorator
- CapitalizeDecorator (and any other decorators)

## Changes Made
- Added comprehensive unit tests for each class and decorator to ensure their methods are functioning correctly.
- Utilized RSpec testing framework for writing and organizing tests.
- Ensured that all test cases pass successfully.

## Testing
- Verified that all unit tests are passing without errors or failures.
- Conducted test runs to confirm that the implemented unit tests provide adequate coverage for the specified classes and decorators.

## Documentation
- Updated README or documentation to reflect the inclusion of unit tests and provide guidance on running them.

## Checklist
- [x] All unit tests have been added for the specified classes and decorators.
- [x] Tests have been successfully run and pass without issues.
- [x] Documentation has been updated to include information about the newly added tests.

---

Regards,

@Zafron047 , @YoutLone

### CODE REVIEW

File: spec/person_spec.rb
Comment (AbelG101) [inline spec/person_spec.rb:1]: - Please make sure to add tests for the following scenarios:
    - [ ] The constructor method should set the default value for the optional name parameter if it is not provided.
    - [ ] For the `add_rental` method, add a new book and check the rentals array of the Person object.

File: spec/capitalize_decorator_spec.rb
Comment (AbelG101) [inline spec/capitalize_decorator_spec.rb:1]: - [ ] Kindly make sure to add tests for the `trimmer_decorator` class as well by passing in long names and checking if it is properly trimmed. 
- [ ] Also please make sure to add tests for the `Teahcer` class as per the project requirements.

**Review by AbelG101 — CHANGES_REQUESTED** [review]
# Hi @Zafron047 & @YoutLone 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
image
- Properly written unit tests ✔
- Proper use of `it` and `describe` block ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with an amazing summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ibizugbe — APPROVED** [review]
Hi @Team👋🏼👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] The required tests has been added as requested by the prrevious reviewer👍🏼.
- [x] All tests are passing👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**